src/dev/build/lib/download.ts

```diff
@@ -83,30 +85,47 @@ export async function downloadToDisk({
       }
 
       const hash = createHash(shaAlgorithm);
-      await new Promise((resolve, reject) => {
+      let bytesWritten = 0;
+
+      await new Promise<void>((resolve, reject) => {
         response.data.on('data', (chunk: Buffer) => {
-          hash.update(chunk);
-          writeSync(fileHandle, chunk);
+          if (!skipChecksumCheck) {
+            hash.update(chunk);
+          }
+
+          const bytes = writeSync(fileHandle, chunk);
+          bytesWritten += bytes;
         });
 
         response.data.on('error', reject);
-        response.data.on('end', resolve);
+        response.data.on('end', () => {
+          if (bytesWritten === 0) {
+            throw new Error(`No bytes written when downloading ${url}`);
```

Comment:
rather than throwing in an error handler I think we should reject with the error